### PR TITLE
fix(tabs): remove max-width

### LINKED ISSUE #11766: Tabs: Remove max-width from Tab
This issue is for simply removing the max-width property from Tab to allow them to flow to 100% the width of their content.

### PR DESCRIPTION
**Pull request by francinelucca**
Closes #11766 

Removes `max-width` property from `Tabs` and removes long label demos from storybook

#### Changelog

**Changed**

- packages/react/src/components/Tabs/Tabs.stories.js: remove tabs stories long labels
- packages/styles/scss/components/tabs/_tabs.scss: remove `max-width`

#### Testing / Reviewing

Override label text in deployment to make it a really long name, see no max-width limitations, compare to https://react.carbondesignsystem.com/?path=/story/components-tabs--default

### CODE REVIEW

**Review by andreancardona — APPROVED** [review]
Not sure why percy was failing but code looks good!

**Review by tay1orjones — APPROVED** [review]

**Comment by tay1orjones** [comment]
As discussed today, maybe we could add a blurb to the storybook .mdx docs for tabs about the max-width. Why it was removed, and what steps users can take to provide a max-width to their tabs.

**Comment by francinelucca** [comment]
> As discussed today, maybe we could add a blurb to the storybook .mdx docs for tabs about the max-width. Why it was removed, and what steps users can take to provide a max-width to their tabs.

Added! @alisonjoseph @andreancardona can you re-rev? 🙏🏻

File: packages/react/src/components/Tabs/Tabs.mdx [hunk 1]
```diff
@@ -13,9 +13,14 @@ import { Tabs, TabList, Tab, TabPanels, TabPanel } from './Tabs';
...
-  - [Container Tabs](#container-tabs)
+  - [Contained Tabs](#contained-tabs)
+  - [Icon Tabs](#icon-tabs)
 - [Component API](#component-api)
-  - [Tab `renderContent`](#tab-rendercontent)
+  - [Tab - render content on click](#tab---render-content-on-click)
+- [V11](#v11)
+  - [Tabs composition](#tabs-composition)
+  - [Various updates](#various-updates)
+  - [Max width](#max-width)
```
Comment (tay1orjones) [inline packages/react/src/components/Tabs/Tabs.mdx:23]: Thanks for running doctoc!

**Review by tay1orjones — APPROVED** [review]

File: packages/react/src/components/Tabs/Tabs.mdx (hunk 1 above)
Comment (francinelucca) [reply packages/react/src/components/Tabs/Tabs.mdx:23]: there's an automated way to do this? 🤣 I wrote it hahahah

Comment (tay1orjones) [reply packages/react/src/components/Tabs/Tabs.mdx:23]: haha, yes! Run `yarn doctoc` in the root or `yarn run doctoc` in a package

**Review by andreancardona — APPROVED** [review]

**Pull request merged**